pkg/bee/client.go

```diff
@@ -223,6 +224,22 @@ func (c *Client) DownloadFile(ctx context.Context, a swarm.Address, opts *api.Do
 	return size, h.Sum(nil), nil
 }
 
+func (c *Client) DownloadActFile(ctx context.Context, a swarm.Address, opts *api.DownloadOptions) (size int64, hash []byte, err error) {
+	r, err := c.api.Act.Download(ctx, a, opts)
+	if err != nil {
+		return 0, nil, fmt.Errorf("download file %s: %w", a, err)
+	}
+	defer r.Close()
```

Comment:
If an error occurs and r is not nil then this line will not be called and 'r' will not close.

Reply:
'r' is of type io.ReadCloser. In case of an error, r will be nil, so Close should not be called to avoid a potential panic.

Reply:
defer r.Close removed

https://github.com/Solar-Punk-Ltd/beekeeper/blob/feat/act/pkg/bee/client.go#L233

Reply:
@ferencsarai sorry for misunderstanding, "defer r.Close()" should be there (like you placed it). I was trying to reply on @martinconic message, but I forgot to tag him.

Reply:
`defer r.Close()` added